engine/schema/src/main/java/com/cloud/storage/dao/VolumeDaoImpl.java

```diff
@@ -217,6 +218,15 @@ public VolumeVO findByPoolIdName(long poolId, String name) {
         return listBy(sc);
     }
 
+    @Override
+    public List<VolumeVO> findReadyAndAllocatedRootVolumesByInstance(long instanceId) {
```

Comment:
nitpick: `findReadyAndAllocatedRootVolumeByInstance` (singular) we are only looking for one volume, can we encounter more?

Reply:
@DaanHoogland not sure but don't we have multi-disk OVA in VMware? Is only a single volume considered ROOT in that case?

Reply:
@DaanHoogland for kvm, there is only 1 root disk. 
for other hypervisors eg vmware, I have no idea. 
so I'd like to use similar code as findReadyRootVolumesByInstance which returns a list of root volumes and processed in other java files.

Reply:
Yes, it is possible to have multiple ROOT disks for deploy-as-is templates on Vmware. However, I'm not seeing a problem as the method is returning a list of volumes